### ISSUE TITLE
need to support distro boot config

### ISSUE THREAD

**Issue by ausil** [MarvellEmbeddedProcessors/u-boot-armada38x#3]
u-boot upstream has support for using a standardised distro mechanism using extlinux.conf files. it needs to be supported

**Comment by qolii** [comment]
Please do this! This is huge.

**Comment by jan-kiszka** [comment]
There are some bits in u-boot-2017.03-armada-18.06 now, but they are incomplete, at least fdtfile is missing. Moreover, it's still based on that old 2017.03 u-boot.

The only reasonable way forward is pushing the remaining bits into upstream (seems like there are problems with RAM size detection and MMC, plus a fatal regression since 2018.07) and finally retiring this fork. That would greatly improve the user experience.

**Comment by kostapr** [comment]
The new Marvell 18.09 release is based on u-boot-2018.03. This code should be available at early September.

**Comment by jan-kiszka** [comment]
Please move to upstream instead. Just started that discussion on the u-boot list. Forks don't work in practice, and that's not only my personal opinion, we face such issues at work all time with vendor trees. I'm happy to explain your management if they think otherwise ;-).

**Comment by kostapr** [comment]
Yes, I understand you, but we have to use some baseline for our QA  and for customers that are going to production. What we have to do is to invest some time in pushing our patches to the mainline tree. This is the step our management gets involved :-)

**Comment by jan-kiszka** [comment]
Eventually, it's just a matter of routing your QA efforts to the right tree (AKA upstream). I know that this may take some time to achieve if your backlog (patch queue) is still too long, but maintaining forks is not the solution. It's only distracting valuable resources.

**Comment by qolii** [comment]
I am super excited for this.

**Comment by kostapr** [comment]
The release 18.09 is out

**Comment by jan-kiszka** [comment]
When will the download page be updated? When do you plan to set fdtfile properly? And when do you have time to follow up on the upstream discussion (https://lists.denx.de/pipermail/u-boot/2018-August/339362.html)?

BTW, see https://github.com/siemens/jailhouse-images/commit/973599edfbf048a9e9afc2342133db99e9e7d415 for the user story when building an image with a real distro for that board. I will redirect our users to the new vendor u-boot once it's available as binary.

**Comment by dhewg** [comment]
Ancient issue, but I opened a PR for OpenWRT to ship a *mainline* u-boot/atf. Which includes distro boot:
https://github.com/openwrt/openwrt/pull/3360

So if anyone is still intrerested, we need testers :)
https://github.com/openwrt/openwrt/pull/3360#issuecomment-683698240
